src/net.cpp

```diff
@@ -2668,6 +2473,25 @@ bool CConnman::DisconnectNode(const std::string& strNode)
     }
     return false;
 }
+
+bool CConnman::DisconnectNode(const CSubNet& subnet)
+{
+    bool disconnected = false;
+    LOCK(cs_vNodes);
+    for (CNode* pnode : vNodes) {
+        if (subnet.Match(static_cast<CNetAddr>(pnode->addr))) {
```

Comment:
Seems `dynamic_cast` would be more appropriate here, given the cast to base class.

Reply:
I'd advise against using `dynamic_cast`. It's not used anywhere else in the code and it relies on C++ run-time type information which has at least historically a large overhead. I'm also fairly sure it's unnecessary here.